doc/i2p.md

```diff
@@ -0,0 +1,64 @@
+# I2P SUPPORT IN BITCOIN
+
+It is possible to use Bitcoin Core to accept and make connections from/to other
+Bitcoin nodes through the [I2P network](https://en.wikipedia.org/wiki/I2P).
+
+## Run Bitcoin Core with an I2P router
+
+A running I2P router (proxy) with [SAM](https://geti2p.net/en/docs/api/samv3)
+enabled is required (there is an [official one](https://geti2p.net) and
+[a few alternatives](https://en.wikipedia.org/wiki/I2P#Routers)). Notice the IP
+address and port the SAM proxy is listening to. Usually that is
+`127.0.0.1:7656`. Once it is up and running with SAM enabled, the following
+Bitcoin Core options can be used to engage with it:
+
+```
+-i2psam=<ip:port>
+     I2P SAM proxy to reach I2P peers and accept I2P connections (default:
+     none)
+
+-i2pacceptincoming
+     If set and -i2psam is also set then incoming I2P connections are
+     accepted via the SAM proxy. If this is not set but -i2psam is set
+     then only outgoing connections will be made to the I2P network.
+     Ignored if -i2psam is not set. Listening for incoming I2P
+     connections is done through the SAM proxy, not by binding to a
+     local address and port (default: 1)
+```
+
+In a typical situation, this suffices:
+
+```
+bitcoind -i2psam=127.0.0.1:7656
+```
+
+The first time Bitcoin Core connects to the I2P router its I2P address (and
+its corresponding private key) will be automatically generated and saved in a
+file named `i2p_private_key` in the Bitcoin Core data directory.
+
+## Additional configuration options related to I2P
+
+You may set the `-debug=i2p` config logging option to have additional
+information in the debug log about your I2P configuration and connections.
+
+It is possible to restrict outgoing connections in the usual way with
+`-onlynet=i2p`. Notice that, I2P support being new (as of 2021), there are not
+as many I2P peers as Tor or IP ones. So using `-onlynet=i2p` may make a node
+more susceptible to
+[Sybil attack](https://en.bitcoin.it/wiki/Weaknesses#Sybil_attack). Use CLI
+`-addrinfo` to see the number of I2P addresses known to your node.
+
+## I2P related information in Bitcoin Core
+
+There are several ways to see your I2P address in Bitcoin Core:
+- in the debug log (grep for `AddLocal`, the I2P address ends in `.b32.i2p`)
+- in the output of RPC `getnetworkinfo` in the "localaddresses" section
+- in the output of the CLI `-netinfo` peer connections dashboard
+
+To see to which I2P peers your node is connected, use `bitcoin-cli -netinfo 4`
+or the RPC `getpeerinfo` (e.g. `bitcoin-cli getpeerinfo`).
+
+## Compatibility
+
+Bitcoin Core uses the [SAM v3.1](https://geti2p.net/en/docs/api/samv3) protocol
+to connect to the I2P network. Any I2P router that supports it can be used.
```

Comment:
Please add a link to this document in `doc/README.md` (the best place is likely where `tor.md` is mentioned) so that people can find it more easily.

Reply:
Done.